pkg/csi/service/common/common_controller_helper.go

```diff
@@ -141,6 +143,69 @@ func CheckAPI(version string) error {
 	return nil
 }
 
+// UseVslmAPIs checks if specified version is between 6.7 Update 3l and 7.0
+// The method takes aboutInfo{} as input which contains details about
+// VC version, build number and so on.
+// If the version is between the upper and lower bounds, the method returns true, else returns false
+// and appropriate errors during failure cases
+func UseVslmAPIs(ctx context.Context, aboutInfo vim25types.AboutInfo) (bool, error) {
+	log := logger.GetLogger(ctx)
+	items := strings.Split(aboutInfo.ApiVersion, ".")
+	apiVersion := strings.Join(items[:], "")
+	// Convert version string to string, Ex: "6.7.3" becomes 673, "7.0.0.0" becomes 700
+	vSphereVersionInt, err := strconv.Atoi(apiVersion[0:3])
+	if err != nil {
+		msg := fmt.Sprintf("Error while converting ApiVersion %q to integer, err %+v", apiVersion, err)
+		log.Errorf(msg)
+		return false, errors.New(msg)
+	}
+	vSphere67u3Version := strings.Join(strings.Split(VSphere67u3Version, "."), "")
+	vSphere67u3VersionInt, err := strconv.Atoi(vSphere67u3Version[0:3])
+	if err != nil {
+		msg := fmt.Sprintf("Error while converting VSphere67u3Version %q to integer, err %+v", VSphere67u3Version, err)
+		log.Errorf(msg)
+		return false, errors.New(msg)
+	}
+	vSphere7Version := strings.Join(strings.Split(VSphere7Version, "."), "")
+	vSphere7VersionInt, err := strconv.Atoi(vSphere7Version[0:3])
+	if err != nil {
+		msg := fmt.Sprintf("Error while converting VSphere7Version %q to integer, err %+v", VSphere7Version, err)
+		log.Errorf(msg)
+		return false, errors.New(msg)
+	}
+	// Check if the current vSphere version is between 6.7.3 and 7.0.0
+	if vSphereVersionInt > vSphere67u3VersionInt && vSphereVersionInt <= vSphere7VersionInt {
+		return true, nil
+	}
+	// Check if version is 6.7.3
+	if vSphereVersionInt == vSphere67u3VersionInt {
+		// CSI migration feature will be supported only from 6.7 Update 3l vSphere version onwards
+		// For all older 6.7 Update 3 such as 3a, 3b and so on, we do not support the CSI migration feature.
+		// Because there is no patch version number in aboutInfo{}, we will rely on the build number to check for 6.7 Update 3l
+		// VC builds are always incremental and hence we can use the build info to check if VC version is 6.7 Update 3l
+		// Here is a snippet of the build info for the GA bits mentioned here: https://docs.vmware.com/en/VMware-vSphere/6.7/rn/vsphere-vcenter-server-67u3l-release-notes.html
+		// Name: "VMware vCenter Server",
+		// FullName: "VMware vCenter Server 6.7.0 build-17137327",
+		// Version: "6.7.0",
+		// Build:  "17137327",
+		// ApiVersion: "6.7.3",
+		if vcBuild, err := strconv.Atoi(aboutInfo.Build); err == nil {
+			if vcBuild >= 17137327 {
```

Comment:
Can we discuss this with the team, if we can rely on the build version? looks like we do not have any option other than relying on the build version for 67u3 releases.

Reply:
> Can you post the result of the new unit test cases on the PR description?

```
common chethanv$ go test -v
=== RUN   TestUseVslmAPIsFuncForVC67Update3l
--- PASS: TestUseVslmAPIsFuncForVC67Update3l (0.00s)
=== RUN   TestUseVslmAPIsFuncForVC67Update3a
{"level":"error","time":"2021-04-01T13:24:48.808725-07:00","caller":"common/common_controller_helper.go:196","msg":"Found vCenter version :\"6.7.3\". The minimum supported vCenter version for CSI migration feature is vCenter Server 6.7 Update 3l.","stacktrace":"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common.UseVslmAPIs\n\t/Users/chethanv/Downloads/csi-external/vsphere-csi-driver/pkg/csi/service/common/common_controller_helper.go:196\nsigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common.TestUseVslmAPIsFuncForVC67Update3a\n\t/Users/chethanv/Downloads/csi-external/vsphere-csi-driver/pkg/csi/service/common/common_controller_helper_test.go:76\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1123"}
--- PASS: TestUseVslmAPIsFuncForVC67Update3a (0.00s)
=== RUN   TestUseVslmAPIsFuncForVC6dot6
--- PASS: TestUseVslmAPIsFuncForVC6dot6 (0.00s)
=== RUN   TestUseVslmAPIsFuncForVC7
--- PASS: TestUseVslmAPIsFuncForVC7 (0.00s)
=== RUN   TestUseVslmAPIsFuncForVC70u1
--- PASS: TestUseVslmAPIsFuncForVC70u1 (0.00s)
```

Reply:
Looks like we agreed on having build number check. Keeping this logic